docs/guide/slots.md

```diff
@@ -145,6 +145,51 @@ end
 <% end %>
 ```
 
+## Referencing slots
+
+You can access the instance of a slot in your ruby file by using the `before_render` [lifecycle method](./lifecycle.md).
+
+```ruby
+# blog_component.rb
+class BlogComponent < ViewComponent::Base
+  renders_one :header
+  renders_one :image
+  renders_many :posts
+
+  def initialize(**system_arguments)
+    @system_arguments = system_arguments
+
+    @post_container_classes = class_names(
+      "PostContainer",
+      system_arguments[:classes]
```

Comment:
Can we simplify a bit by removing `system_arguments`?


```suggestion
  def initialize(classes: [])
    @post_container_classes = class_names("post-container", classes)
```

What do you think?

Reply:
Absolutely!